torch/nn/functional.py

```diff
@@ -406,6 +407,28 @@ def dropout(input, p=0.5, training=False, inplace=False):
     return _functions.dropout.Dropout(p, training, inplace)(input)
 
 
+def alpha_dropout(input, p=0.5, training=False):
+    if not 0 < p <= 1:
+        raise ValueError("dropout probability has to be between 0 and 1, "
+                         "but got {}".format(p))
+
+    if p > 0 and training:
+        alpha = -1.7580993408473766
+        keep_prob = 1 - p
+        noise = input.data.new(input.size())
+        noise.bernoulli_(keep_prob)
+        noise = Variable(noise)
+
+        output = (input * noise).add_(noise.neg().add_(1).mul_(alpha))
```

Comment:
is it worth compromising readability in this case for smaller memory footprint? I suppose with jit compilation these point-wise expressions will be compiled to a single kernel anyway.